### PR TITLE
ui for closing notifications closes #1870

### LINKED ISSUE #1870: Error message should have close button to dismiss the message
- Apostrophe modal error messages currently doesn't have a close button to let the back end users dismiss these errors and it will never go away unless you reload the page

- Currently the only way to dismiss/acknowledge this error message is to click on the message itself but how will the back end user know that ? 

- Please add a close button to allow the back end user to dismiss this error message

- This is a sample override code to change the error messages in custom coded module/piece:

```
// Inside lib/modules/apostrophe-users/public/js/editor-modal.js
apos.define('apostrophe-users-editor-modal', {

  extend: 'apostrophe-pieces-editor-modal',

  construct: (self, options) => {
    self.getErrorMessage = function (err) {
      if (err === 'unique') {
        return 'Please make sure the username and email address fields are unique.';
      } else if (err === 'Invalid Password') {
        return 'Invalid Password!';
      } else if (err === 'Invalid Administrator Password') {
        return 'Invalid Administrator Password!';
      }
      else {
        return 'An error occurred. Please try again.';
      }
    };
  }
});
```

### PR DESCRIPTION
**Pull request by boutell**
I'm just a backend hacker but I think this came out surprisingly unterrible:

Screen Shot 2019-09-02 at 3 08 07 PM

What say you @stuartromanek?

### CODE REVIEW

**Comment by boutell** [comment]
See also #1845, confusion about how to dismiss these is common.

**Comment by abea** [comment]
We are missing some easy a11y here, including in the existing code. The `span` should definitely be a `button`, for this contrib.

We should also be using the `alert` role on the whole thing. That would mean moving from only an opacity change to a `display` change, though, so maybe to save for another PR.

File: lib/modules/apostrophe-notifications/views/notification.html [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <div class="apos-ui apos-notification apos-notification--hidden" data-apos-notification>
-  <p class="apos-notification-message" data-apos-notification-message>{{ data.message }}</p>
+  <p class="apos-notification-message" data-apos-notification-message>{{ data.message }}<span class="apos-notification-message-close"><i class="fa fa-times-circle" aria-hidden="true"></i></span></p>
```
Comment (abea) [inline lib/modules/apostrophe-notifications/views/notification.html]: We should use a `button` here for semantics and the built-in benefits. Or maybe the `div` around the whole thing should be a `button` if clicking the whole thing closes it. That'd only be bad if auto-dismiss ones don't dismiss on click too.

Comment (abea) [reply lib/modules/apostrophe-notifications/views/notification.html]: Holler if you want me to take a swipe at that.

Comment (boutell) [reply lib/modules/apostrophe-notifications/views/notification.html]: They all dismiss on click. Feel free to make it a button and get the styles happy with that. Thanks.

Comment (boutell) [reply lib/modules/apostrophe-notifications/views/notification.html]: Agree we should do what's expedient in this PR (fill the current UX hole) and continue to iterate. Button seems reasonable to do now.

**Comment by abea** [comment]
Updated. @stuartromanek Could you take a look since I did a bunch in there?

**Pull request merged**